src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -200,6 +226,27 @@ public StructType(IEnumerable<StructField> fields)
                 new JProperty("type", TypeName),
                 new JProperty("fields", Fields.Select(f => f.JsonValue).ToArray()));
 
+        /// <summary>
+        /// Does this type need to convert between C# object and internal SQL object.
+        /// </summary>
+        /// <returns>True if conversion needed, false otherwise</returns>
+        internal override bool NeedConversion() => true;
+
+        /// <summary>
+        /// Converts an internal SQL object into a native C# object.
+        /// </summary>
+        /// <param name="obj">The object to convert.</param>
+        /// <returns>Converted C# object</returns>
+        internal override object FromInternal(object obj)
+        {
+            if ((obj is null) || (obj is Row))
+            {
+                return obj;
+            }
+
+            throw new NotImplementedException();
```

Comment:
is this case `NotImplmentedException`? or something not expected?

Reply:
Removed interfaces and reverted code.